tesseract_motion_planners/core/include/tesseract_motion_planners/core/planner.h

```diff
@@ -79,7 +79,7 @@ class MotionPlanner
                              bool format_result_as_input);
 
 protected:
-  std::string name_;
+  std::string ns_;
 };
 }  // namespace tesseract_planning
```

Comment:
Although at the task composer level we will be using the namespace as the name, I still think this should remain the name.

Reply:
I agree, changing `name_` to `ns_` feels weird. The reason I changed this is because it makes getting profiles feel inconsistent otherwise. 

By keeping `name_` In the `MotionPlanner` base class object, in all inherited Tasks, like Descartes or OMPL or TrajOpt, you'd have a line like this:
```C++
auto cur_plan_profile = getProfile<DescartesPlanProfile<FloatType>>(
  name_, profile, *request.profiles, std::make_shared<DescartesDefaultPlanProfile<FloatType>>());
```
But in something like `MinLengthTask` that inherits from `TaskComposerTask` you have:
```C++
auto cur_composite_profile =
      getProfile<MinLengthProfile>(ns_, profile, *problem.profiles, std::make_shared<MinLengthProfile>());
```
In one example you reference the `name_` and in the other you reference `ns_`, and unless you know the history of all of this that might be really confusing.


The other idea I had was to just make a method in the `MotionPlanner` base class and the `TaskComposerTask` base class that would be something like
```C++
template <typename ProfileType>
td::shared_ptr<const ProfileType> getProfile(const std::string& profile, const ProfileDictionary& profile_dictionary)
{
  return getProfile<ProfileType>(<ns_ or name_>, profile, profile_dictionary, std::make_shared<ProfileType>());
}
```
Then in both `MotionPlanner` and `TaskComposerTask` inherited classes you can just call
```C++
getProfile<DescartesPlanProfile<FloatType>>(profile, *request.profiles);
```
or
```C++
getProfile<MinLengthProfile>(profile, *problem.profiles);
```
And then we can replace all instances of getting the profile with a less confusing interface, and I can revert `ns_` back to `name_`

Reply:
I don't think all of that is necessary, because they two different packages that use profiles. One way could be to just document why motion planners are using its name. Another option is to update PlannerRequest to include the profile namespace parameter. Then in the arm planner have something like.

```
const std::string profile_ns = (request.profile_ns.empty() ? name_ : request.profile_ns;
```

Reply:
Okay, I reverted the change of `name_` to `ns_`.

> `const std::string profile_ns = (request.profile_ns.empty() ? name_ : request.profile_ns;`

I think that might make it more confusing.